mathics/builtin/algebra.py

```diff
@@ -1413,3 +1413,202 @@ def apply(self, expr, form, h, evaluation):
+                    elif match(factor, target_pat, evaluation):
+                        powers.append(factor)
+                    elif (
+                        factor.has_form("Power", 2) or factor.has_form("Sqrt", 1)
+                    ) and match(factor._leaves[0], target_pat, evaluation):
+                        powers.append(factor)
+                    else:
+                        coeffs.append(factor)
+            else:
+                coeffs.append(term)
+            # Now, rebuild both factors
+            if len(coeffs) == 0:
+                coeffs = None
+            elif len(coeffs) == 1:
+                coeffs = coeffs[0]
+            else:
+                coeffs = Expression("Times", *coeffs)
+            if len(powers) == 0:
+                powers = None
+            elif len(powers) == 1:
+                powers = powers[0]
+            else:
+                powers = Expression("Times", *sorted(powers))
+            return coeffs, powers
+
+        if expr.is_free(target_pat, evaluation):
+            if filt:
+                return Expression(filt, expr).evaluate(evaluation)
+            else:
+                return expr
+        elif expr.is_symbol() or expr.has_form("Power", 2) or expr.has_form("Sqrt", 1):
+            if filt:
+                return Expression(
+                    "Times", Expression(filt, Integer1).evaluate(evaluation), expr
+                )
+            else:
+                return expr
+        elif expr.has_form("Plus", None):
+            coeff_dict = {}
+            powers_dict = {}
+            powers_order = {}
+            for term in expr._leaves:
+                coeff, powers = split_coeff_pow(term)
+                pl = powers_list(powers)
+                key = str(pl)
+                if not key in powers_dict:
+                    powers_dict[key] = powers
+                    coeff_dict[key] = []
+                    powers_order[key] = key_powers(pl)
+
+                coeff_dict[key].append(Integer1 if coeff is None else coeff)
+
+            terms = []
+            for key in sorted(
+                coeff_dict, key=lambda kv: powers_order[kv], reverse=False
+            ):
+                val = coeff_dict[key]
+                if len(val) == 0:
+                    continue
+                elif len(val) == 1:
+                    coeff = val[0]
+                else:
+                    coeff = Expression("Plus", *val)
+                if filt:
+                    coeff = Expression(filt, coeff).evaluate(evaluation)
+
+                powerfactor = powers_dict[key]
+                if powerfactor:
+                    terms.append(Expression("Times", coeff, powerfactor))
+                else:
+                    terms.append(coeff)
+
+            return Expression("Plus", *terms)
+        else:
+            if filt:
+                return Expression(filt, expr).evaluate(evaluation)
+            else:
+                return expr
+
+
+# tejimeto
```

Comment:
Google translate doesn't recognize this as a Spanish word.

Reply:
ja! Sorry, that was a personal note, about a  word my daughter invented...